static/js/hotkey.js

```diff
@@ -37,12 +39,28 @@ import * as stream_popover from "./stream_popover";
 import * as subs from "./subs";
 import * as topic_zoom from "./topic_zoom";
 import * as ui from "./ui";
+import * as unread from "./unread";
 
 function do_narrow_action(action) {
     action(message_lists.current.selected_id(), {trigger: "hotkey"});
     return true;
 }
 
+function check_unread_message() {
+    const last_user_msg = message_lists.current.get_last_message_sent_by_me();
+
+    if (last_user_msg && narrow_state.operators().length === 1) {
+        const stream_unread_messages = unread.get_msg_ids_for_stream(last_user_msg.stream_id);
```

Comment:
This should use `current_msg_list` as the source for messages; that'll work with any view, not just stream narrows.

And then I think we can just iterate through the messages in that message list and check if the last message sent by you is above or below the last unread message in that view.